webapp/src/components/StatusCheck.tsx

```diff
@@ -1,14 +1,44 @@
-import { Box, Typography } from "@mui/material";
+import { Box, capitalize, Stack, Typography } from "@mui/material";
 import noData from "assets/launch.svg";
 import React from "react";
 import { useParams } from "react-router-dom";
 import { getStatusEndpoint } from "services/api";
 import Loading from "./Loading";
+import Grid from "@mui/material/Grid";
+import Paper from "@mui/material/Paper";
+import CheckIcon from "./Icons/Check";
+import ErrorIcon from "@mui/icons-material/Error";
+import PendingOutlinedIcon from "@mui/icons-material/PendingOutlined";
 
 type Props = {
   children: React.ReactNode;
 };
 
+interface StringByString {
+  [key: string]: any;
+}
+
+const STATUS_ICON_MAPPING: StringByString = {
+  finished: <CheckIcon />,
+  not_started: <CheckIcon />,
```

Comment:
It's a detail, but I would suggest using the `Done` icon from MUI. It is the same as `Check`, but it seems semantically perfect in this context to use this icon when the task is "done".
```ts
import DoneIcon from "@mui/icons-material/Done";
```
and something like
```suggestion
  finished: <DoneIcon color="success" />,
  not_started: <DoneIcon color="success" />, // The task result was already cached, so the task didn't need to be started. 
```